src/utils/gadm_gpkg_processing.py

```diff
@@ -0,0 +1,200 @@
+    # Last list element, Last character
+    last_populated_gid = gdf_gid.columns.values.tolist()[-1][-1]
+    print('\tMax Admin Level {0}'.format(last_populated_gid))
+    # Drop all cols after first empty GID_* column.
+    drop_cols = []
+    for col in cols:
+        if is_int(col[-1]) is True and int(col[-1]) > int(last_populated_gid):
+            drop_cols.append(col)
+    gdf.drop(drop_cols, axis=1, inplace=True)
+    # and take out ID_* fields
+    gdf.drop(gdf[fnmatch.filter(gdf.columns.values.tolist(), 'ID_*')], 
+             axis=1, inplace=True)
+    # and take out REMARKS_*
+#    gdf.drop(gdf[fnmatch.filter(gdf.columns.values.tolist(), 'REMARKS_*')], 
+#             axis=1, inplace=True)
+    # and take out VALID*
+#    gdf.drop(gdf[fnmatch.filter(gdf.columns.values.tolist(), 'VALID*')], 
+#             axis=1, inplace=True)
+    # and the UID
+    gdf.drop(columns=['UID'], axis=1, inplace=True)
+    # anything else is associated with the last avialable admin level if
+    # populated.
+    gdf.dropna(axis=1, how='all', inplace=True)
+    #
+    level = 0
+    while level < int(last_populated_gid):
+        print('\tProcessing Level {0}'.format(level))
+        cols_agg = []
+        cols_required = []
+        for col in gdf.columns.values.tolist():
+            if is_int(col[-1]) is True and int(col[-1]) <= level:
+                cols_agg.append(col)
+        # Include the geometry column 
+        cols_required = list(cols_agg)
+        cols_required.append('geometry')
+        # Dissolve using GID_* only - assume all other fields to use First
+        # value. Was dropping features where looking to be specific and
+        # groupby more fields (that should all be the same)
+        cols_agg = [x for x in cols_agg if 'GID_' in x]
+        gdf_dslv = gdf[cols_required].dissolve(by=cols_agg)
+        gdf_dslv.reset_index(inplace=True)
+        gdf_dslv = gdf_dslv[cols_required]
+        print('\tAggregated to {0} features.'.format(len(gdf_dslv)))
+        return_dict["a{0}".format(level)] = gdf_dslv 
+        # 
+        level += 1
+    # Include the highest level
+    return_dict["a{0}".format(last_populated_gid)] = gdf
+    return return_dict
+        
+    
+##
+# If gdf is passed only no way to know what the AOI country is.
+def get_neighbour_countries(gdf_source, gdf_A0=None, NAME_0=None, GID_0=None):
+    print('Extending bounding box.')
+    # https://stackoverflow.com/questions/19412462/getting-distance-between-two-points-based-on-latitude-longitude#19412565
+    bbox = gdf_A0.bounds
+    minx = bbox.minx[0]
+    maxx = bbox.maxx[0]
+    miny = bbox.miny[0]
+    maxy = bbox.maxy[0]
+    # Ropey way to scale the WGS84 bounding box by a factor... 
+    # Should probably map these to see what the distortion ends up as.
+    factor = 0.1    # does this mean 10% ???
+    pcLon = abs(minx-maxx) / 360    # Total degrees X axis
+    lonDistScaled = 360 * ((pcLon * factor) + pcLon) 
```

Comment:
This is equivalent to: 
`longDistScaled = abs(minx-maxxx) * (factor + 1)`
So I don't think there is a need to scale by degrees as it just drops out (same with for the latitude)